### DISCUSSION TITLE
JavaScript ☕ Programming Language Thread

### DISCUSSION THREAD

**Discussion by queenofcorgis** [community/community#51389] [Programming Help]
JavaScript(JS) is a programming language known for creating dynamic content and is used by over 95% of websites around the world.  _$ denotes a paid resource._ 

**Around the web**

- [Javascript Algorithms](https://github.com/trekhleb/javascript-algorithms) 
- [30 Second of Code](https://github.com/30-seconds/30-seconds-of-code) 
- [Intro to JavaScript](https://www.codecademy.com/learn/introduction-to-javascript) 
- [The Odin Project - Full Stack JavaScript](https://www.theodinproject.com/paths/full-stack-javascript?)
- [Codecademy - Learn JavaScript](https://www.codecademy.com/learn/introduction-to-javascript)
- $ [Learn Enough JavaScript to Be Dangerous](https://www.learnenough.com/javascript) 

**Books**
- JavaScript: The Good Parts - Douglas Crockford

Share any links, tutorials, videos, and books that have helped you learn JavaScript below - don’t forget to include if it is paid too!

**Comment by sergeyleschev** [comment]
Online Tutorials:
[Codecademy - Learn JavaScript](https://www.codecademy.com/learn/introduction-to-javascript)
[FreeCodeCamp - JavaScript Algorithms and Data Structures Certification](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/)
[JavaScript.info](https://javascript.info/)

YouTube Channels:
[The Net Ninja](https://www.youtube.com/c/TheNetNinja)
[Traversy Media](https://www.youtube.com/user/TechGuyWeb)
[Programming with Mosh](https://www.youtube.com/user/programmingwithmosh)

Books:
[Eloquent JavaScript by Marijn Haverbeke](https://eloquentjavascript.net/)
[You Don't Know JS by Kyle Simpson](https://github.com/getify/You-Dont-Know-JS/tree/2nd-ed)
JavaScript: The Good Parts by Douglas Crockford

**Reply by queenofcorgis** [reply]
Woah - this is awesome, thanks for sharing so many great options @sergeyleschev ❤️ .

**Reply by Lobonjulian** [reply]
falto [The Odin](https://www.theodinproject.com/)

**Comment by RumeysaaKopuz** [comment]
Thanks for the resources.

**Comment by onlinecare078** [comment]
Hello friends am new to web development, 
can u help and give me a guide or part way i should follow so that i can achieve to be a full web developer on both frontend and backend development.
I want to be able to code, read and debug code.
Thanks
